### ISSUE TITLE
Migration to version 10.0

### ISSUE THREAD

**Issue by pedrobaeza** [OCA/account-invoicing#194]
# Todo

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0
# Modules to migrate
- [x] account_group_invoice_lines - By @alexis-via - #208 
- [x] account_invoice_blocking - By @adrienpeiffer - #197 
- [x] account_invoice_check_total - By @adrienpeiffer - #195 
- [ ] account_invoice_customer_ref_unique - By @aheficent - #257
- [x] account_invoice_force_number - By @tafaRU  - #242
- [x] account_invoice_line_description - By @luismontalba - #274 
- [ ] account_invoice_line_sort
- [x] account_invoice_merge - By @lreficent - #238 
- [x] account_invoice_merge_payment - By @benwillig - #264 
- [x] account_invoice_merge_purchase - By @adrienpeiffer - #251 
- [ ] account_invoice_partner
- [ ] account_invoice_period_usability
- [x] account_invoice_pricelist - By @rvalyi - #342 
- [x] account_invoice_pricelist_sale - By @rvalyi - #343 
- [x] account_invoice_refund_link - By @luismontalba - #254 
- [x] account_invoice_refund_option - ~~By @NeoPolus - #229~~ Not needed in v10
- [x] account_invoice_rounding - By @grindtildeath - #212
- [x] ~~account_invoice_shipping_address~~ - Feature included in core
- [x] account_invoice_supplier_ref_unique - By @adrienpeiffer - #192 
- [x] account_invoice_supplierinfo_update - By @mourad-ehm - #252
- [x] account_invoice_supplierinfo_update_discount - By @legalsylvain - #468
- [ ] account_invoice_template - By @jzahid - #250 
- [ ] account_invoice_uom
- [ ] account_invoice_validation_workflow
- [ ] account_invoice_zero_autopay
- [x] account_payment_term_extension - By @alexis-via - #193 
- [x] invoice_fiscal_position_update > account_invoice_fiscal_position_update - By @alexis-via - #191 
- [ ] product_customer_code_invoice - By @angedras - #267 
- [ ] purchase_batch_invoicing - By @Montxu - #456 
- [ ] purchase_stock_picking_return_invoicing
- [ ] sale_order_partial_invoice
- [x] sale_timesheet_invoice_description - By @pedrobaeza - #300 
- [ ] stock_invoice_picking_incoterm
- [ ] stock_picking_invoicing

**Comment by leemannd** [comment]
@pedrobaeza Port of account_invoice_rounding here https://github.com/OCA/account-invoicing/pull/212

**Comment by pedrobaeza** [comment]
OK

**Comment by leemannd** [comment]
@pedrobaeza Sorry, it's https://github.com/grindtildeath/ who is doing it.

**Comment by pedrobaeza** [comment]
OK, fixed

**Comment by misern2** [comment]
@pedrobaeza I think that  account_invoice_zero_autopay module is no longer needed in V10.

**Comment by pedrobaeza** [comment]
@misern2 can you point to the code in Odoo that makes that?

**Comment by misern2** [comment]
Reviewing the code, I found out the reconciled field in the account model. This field is a computed field according the following function : 

 https://github.com/odoo/odoo/blob/ea2faefe0d6a4a6063d48fc2cda8d7071d9eefa4/addons/account/models/account_invoice.py#L88-L107

After that, the _write function marks as paid all the reconcilied invoices.

https://github.com/odoo/odoo/blob/ea2faefe0d6a4a6063d48fc2cda8d7071d9eefa4/addons/account/models/account_invoice.py#L352-L361


Am I Right or I'm missing something  ?

**Comment by Jerther** [comment]
I believe **invoice_shipping_address** should be obsolete and not migrated in v10 (v9 also?) since Odoo can already do this. See [here](https://www.odoo.com/documentation/user/9.0/sales/quotation/setup/different_addresses.html).

**Comment by pedrobaeza** [comment]
@Jerther it's not the same feature. Please check module readme.

**Comment by Jerther** [comment]
Oh, you're right. The field itself, Odoo takes care of it but I missed the sale.py and stock.py modules which could be migrated indeed.

**Comment by angedras** [comment]
Started migration of product_customer_code_invoice for 10.0
Depends on: [#236](https://github.com/OCA/product-attribute/pull/236)

**Comment by AaronHForgeFlow** [comment]
account_invoice_customer_ref_unique: https://github.com/OCA/account-invoicing/pull/257

**Comment by ThomasBinsfeld** [comment]
@pedrobaeza account_invoice_merge is merged (#238)

**Comment by pedrobaeza** [comment]
Right, updated!

**Comment by Montxu** [comment]
purchase_batch_invoicing #456

**Comment by legalsylvain** [comment]
account_invoice_supplierinfo_update_discount available here.
https://github.com/OCA/account-invoicing/pull/468
